mlflow/ml-package-versions.yml

```diff
@@ -66,63 +66,30 @@ tensorflow:
       pip install tf-nightly
 
   models:
-    minimum: "2.3.0"
+    minimum: "2.6.0"
     maximum: "2.9.1"
     requirements:
-      "< 2.6.0": ["protobuf<4.0.0"]
       # Requirements to run tests for keras
-      "== dev": ["scikit-learn", "pyspark", "pyarrow"]
+      ">= 0.0.0": ["scikit-learn", "pyspark", "pyarrow"]
     run: |
-      pytest tests/tensorflow/test_tensorflow2_model_export.py tests/tensorflow/test_tensorflow2_metric_value_conversion_utils.py
-
-      # Run tests for keras against keras-nightly that's installed with tf-nightly:
-      # https://github.com/tensorflow/tensorflow/blob/v2.6.0/tensorflow/tools/pip_package/setup.py#L110-L122
-      if [ "$PACKAGE_VERSION" == "dev" ]; then
-        pytest tests/keras/test_keras_model_export.py
-      fi
+      tf_version=$(pip freeze | grep 'tensorflow==' | grep -Eo '[0-9].[0-9]')
+      pip install keras==${tf_version} tensorflow-estimator==${tf_version}
```

Comment:
In which tensorflow version do we need keras reinstallation?

Reply:
If the problematic version is only 2.6.0, we can just use the `requirements` field.

Reply:
For example, `install_requires` in TF 2.6.5 looks like this:

https://github.com/tensorflow/tensorflow/blob/v2.6.5/tensorflow/tools/pip_package/setup.py#L114-L115

```
    'tensorflow_estimator >= 2.6.0, < 2.7',
    'keras >= 2.6.0, < 2.7',
```

`tensorflow_estimator` and `keras` version are restricted to compatible versions.

Reply:
TF 2.6.0:

https://github.com/tensorflow/tensorflow/blob/v2.6.0/tensorflow/tools/pip_package/setup.py#L105-L106

```
    'tensorflow_estimator ~= 2.6',
    'keras ~= 2.6',
```

Reply:
@WeichenXu123 Can we try to run tests with TF < 2.6 and see if we can support TF >= 2.0 instead of TF >= 2.6? If we can't, it's okay to limit the support to TF >= 2.6.

Reply:
@harupy 

I removed "reinstall keras" command here and let's see cross test run results. I remember previous run failed because of incompatible keras version installed, e.g., TF 2.6.0 with keras 2.9.0, not sure why.